lib/std/crypto/gimli.zig

```diff
@@ -291,15 +295,19 @@ pub const Aead = struct {
             in = in[State.RATE..];
             out = out[State.RATE..];
         }) {
-            for (buf[0..State.RATE]) |*p, i| {
-                out[i] = p.* ^ in[i];
-                p.* = in[i];
+            const d = in[0..State.RATE].*;
```

Comment:
The pattern used in the other PR (and below) is slightly cheaper as it avoids this 16-byte copy and keeps a single loop.

Reply:
Having two loops looks more efficient.

With a single loop, the buffer is updated byte by byte. With two loops, the second one is replaced by a 128-bit move.
The 16 byte copy is also just a 128-bit move, whereas the previous pattern requires a save to a temporary register for every byte.

Reply:
Duh, I expected the vectorizer to kick in anyway. If you checked the emitted instructions carry on then.
(Is it worth to apply the same fix below, in the tail processing?)

Reply:
In tail processing, the size is not constant, so the loop cannot be optimized away.